code/modules/mob/living/simple_animal/animal_defense.dm

```diff
@@ -16,7 +16,23 @@
 		if("grab")
 			grabbedby(M)
 
-		if("harm", "disarm")
+		if("disarm")
+			M.do_attack_animation(src, ATTACK_EFFECT_DISARM)
+			playsound(src, 'sound/weapons/thudswoosh.ogg', 50, TRUE, -1)
+			var/shove_dir = get_dir(M, src)
+			if(!Move(get_step(src, shove_dir), shove_dir))
+				log_combat(M, src, "shoved", "failing to move it")
+				visible_message("<span class='danger'>[M.name] shoves [src]!</span>",
+					"<span class='danger'>You shove [src]!</span>", "<span class='hear'>You hear aggressive shuffling!</span>", COMBAT_MESSAGE_RANGE, src)
+				to_chat(src, "<span class='userdanger'>You're shoved by [M.name]!</span>")
+				return TRUE
+			log_combat(M, src, "shoved", "pushing it")
+			visible_message("<span class='danger'>[M.name] shoves [src], pushing [p_them()]!</span>",
+				"<span class='danger'>You shove [src], pushing [p_them()]!</span>", "<span class='hear'>You hear aggressive shuffling!</span>", COMBAT_MESSAGE_RANGE, src)
```

Comment:
visible_message() sends self_message (the second arg) to the mob it's called on, in this case the poor little robot/cat. So I the player see Me pushed Src, pushing it, and the mob would see You shove Yourself, pushing it. 

If you wanted the same behavior you could call visible_message() on M, the human who did it, or you could change the message to reflect who it's being sent to. Up to you.

Reply:
You are right, that wouldn't have worked as intended. Good catch.